prac_07/project_management.py

```diff
@@ -0,0 +1,172 @@
+                if filter_date <= project.start_date:
+                    filtered_project_information += [project]
+            display_project_information(filtered_project_information, choice)
+
+        elif choice == "A":
+            new_project = add_new_project()
+            project_information.append(new_project)
+            print(f"Project: '{new_project.name}, "
+                  f"start: {new_project.start_date}, "
+                  f"priority = {new_project.priority}, "
+                  f"estimate: ${new_project.cost_estimate}, "
+                  f"completion: {new_project.completion_percentage}%' has been added")
+
+        elif choice == "U":
+            display_project_information(project_information, choice)
+            project_information, completed_projects, incomplete_projects = update_project_information(
+                project_information)
+
+        else:
+            print('Invalid menu choice')
+
+        print(MENU)
+        choice = input(">>> ").upper()
+
+    save_to_file(project_information)  # save to file when quitting
+    print("The file has been saved")
+    print("The program has ended")
+
+
+def display_project_information(project_information, choice):
+    """Print a sorted and formatted list of the projects"""
+    completed_projects = []
+    incomplete_projects = []
+    sorted_project_information = sorted(project_information)
+    for project in sorted_project_information:
+        if int(project.completion_percentage) == 100:
+            completed_projects.append(project)
+        else:
+            incomplete_projects.append(project)
+    if choice == 'D':
+        print("Incomplete projects:")
+        for project in incomplete_projects:
+            formatted_date = project.start_date.strftime('%d/%m/%Y')
+            print(f"{project.name}, start: {formatted_date}, priority = {project.priority}, "
+                  f"estimate: ${project.cost_estimate}, completion: {project.completion_percentage}%")
+
+        print("Completed projects:")
+        for project in completed_projects:
+            formatted_date = project.start_date.strftime('%d/%m/%Y')
+            print(f"{project.name}, start: {formatted_date}, priority = {project.priority}, "
+                  f"estimate: ${project.cost_estimate}, completion: {project.completion_percentage}%")
+    else:
+        entry_number = 0
+        for project in sorted_project_information:
+            entry_number += 1
+            formatted_date = project.start_date.strftime('%d/%m/%Y')
+            print(f"{entry_number} - {project.name}, start: {formatted_date}, priority = {project.priority}, "
+                  f"estimate: ${project.cost_estimate}, completion: {project.completion_percentage}%")
+    return sorted_project_information
+
+
+def read_from_file():
+    """Read project information from the file specified by the user"""
+    project_information = []
+    completed_projects = []
+    incomplete_projects = []
```

Comment:
Why are these lists being defined here? Can you make you of a method within your class to do this anyway?